embedded-graphics/src/style/primitive_style.rs

```diff
@@ -79,6 +79,63 @@ where
 }
 
 /// Primitive style builder.
+///
+/// Use this builder to create styles for [primitives]. For convenience, the [`primitive_style!`]
+/// macro is also provided.
+///
+/// If any properties on the builder are ommitted, the value will default to `None` (transparent).
+///
+/// # Examples
+///
+/// ## Build a style with customised stroke and fill
+///
+/// This example builds a style for a circle with a 3px red stroke and a solid green fill. The
+/// circle is centered at (20, 20) with a radius of 10px.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     egtext,
+///     pixelcolor::Rgb565,
+///     primitives::Circle,
+///     prelude::*,
+///     style::{PrimitiveStyle, PrimitiveStyleBuilder},
+/// };
+///
+/// let style: PrimitiveStyle<Rgb565> = PrimitiveStyleBuilder::new()
+///     .stroke_color(Rgb565::RED)
+///     .stroke_width(3)
+///     .fill_color(Rgb565::GREEN)
+///     .build();
+///
+/// let circle = Circle::new(Point::new(20, 20), 10)
+///     .into_styled(style);
+/// ```
+///
+/// ## Build a style with ommitted (transparent) values
```

Comment:
IMO the headline should contain what the example tries to achieve and not how this is done. Something like:
`Build a style with stroke and without fill`.
The description below should then provide how this is achieved.